### ISSUE TITLE
Netty incorrectly strips whitespace from the beginning of the first header name

### ISSUE THREAD

**Issue by kenballus** [netty/netty#14177]
### Steps to reproduce
1. Start a Netty HTTP/1.1 server that echos the parsed components of the request, such as [this one](https://raw.githubusercontent.com/narfindustries/http-garden/main/images/netty/GardenServer.java).
2. Send it a request in which the first header is a Content-Length header prefixed with a whitespace character:
```sh
(printf 'POST / HTTP/1.1\r\n\tContent-Length: 1\r\n\r\nX'; sleep 1) \
    | ncat localhost 80 \
    | grep '"body"' \
    | jq '.["body"]' \
    | xargs echo \
    | base64 -d \
    | xxd
```
3. Observe that a message body is present, indicating that the `Content-Length` header was interpreted, despite the leading tab:
```
00000000: 58                                       X
```

### Expected behavior
The request is invalid and should be rejected with a 400 response. This is what most HTTP implementations do, including AIOHTTP, Apache httpd, Cheroot, FastHTTP, Go net/http, Gunicorn, H2O, HAProxy, Hyper, Hypercorn, Jetty, Libevent, Lighttpd, Nginx, Node.js LiteSpeed, Passenger, Puma, Tomcat, Uvicorn, Waitress, WEBrick, OpenBSD httpd, Apache Traffic Server, nghttpx, Pingora, Varnish, Akamai GHost, Google Classic App LB, Envoy, and OpenBSD relayd.

### Actual behavior
Netty strips whitespace from the beginning of the first request header.

### Netty version
5.1 @ ec867b6ee170f0c8872bd8f033c6d2aef2163961

### JVM version (e.g. `java -version`)
```
openjdk version "23-ea" 2024-09-17
OpenJDK Runtime Environment (build 23-ea+20-Debian-1)
OpenJDK 64-Bit Server VM (build 23-ea+20-Debian-1, mixed mode, sharing)
```

### OS version (e.g. `uname -a`)
```
Linux 416cc96d1b8b 6.9.7-arch1-1 #1 SMP PREEMPT_DYNAMIC Fri, 28 Jun 2024 04:32:50 +0000 x86_64 GNU/Linux
```

**Comment by normanmaurer** [comment]
Should be fixed by https://github.com/netty/netty/pull/14179 @kenballus

**Comment by kenballus** [comment]
Confirmed; this is fixed by #14179.
